### PR TITLE
Improving Matrix3d.ts: Part 4 (src - line 2300 to 2740 (end))

### PR DESCRIPTION
**Pull request by saeeedtorabi**
- improved src code formatting
- improved documentation

### CODE REVIEW

File: docs/learning/geometry/Angle.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ## fraction of an angle sweep returned by different API
-![>](./figs/Angle/angleSweepFraction.jpg)
+![>](./figs/Angle/AngleSweepFraction.jpg)
```
Comment (pmconne) [inline docs/learning/geometry/Angle.md:2]: What was the motivation for this change? Filenames are case-sensitive on some platforms and the case here no longer matches.

File: docs/learning/geometry/Angle.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 
 ## half-angle using the dot products to find ellipse semi-axis
-![>](./figs/Angle/ellipseHalfAngle.jpg)
+![>](./figs/Angle/EllipseHalfAngle.jpg)
```
Comment (pmconne) [inline docs/learning/geometry/Angle.md:5]: Same question about filename case change.

File: docs/learning/geometry/Angle.md (hunk 1 above)
Comment (saeeedtorabi) [reply docs/learning/geometry/Angle.md:2]: That's because I noticed all image names under docs/learning/geometry are CamelCase expect these 2 files

Comment (pmconne) [reply docs/learning/geometry/Angle.md:2]: Do you intend to rename the files, or to revert the changes to the markdown?

**Review by pmconne — APPROVED** [review]

File: core/geometry/src/geometry3d/Matrix3d.ts [hunk 3]
```diff
@@ -2525,86 +2552,105 @@ export class Matrix3d implements BeJSONFunctions {
...
   public get isUpperTriangular(): boolean {
     const sumAll = this.sumSquares();
     const sumLow = Geometry.hypotenuseSquaredXYZ(this.coffs[3], this.coffs[6], this.coffs[7]);
     return Math.sqrt(sumLow) <= Geometry.smallAngleRadians * (1.0 + Math.sqrt(sumAll));
   }
-  /** If the matrix is diagonal and all diagonals are within tolerance, return the first diagonal.  Otherwise return undefined.
+  /** Test if the above diagonal entries (1,2,5) are all nearly zero */
+  public get isLowerTriangular(): boolean {
+    const sumAll = this.sumSquares();
+    const sumLow = Geometry.hypotenuseSquaredXYZ(this.coffs[1], this.coffs[2], this.coffs[75]);
```
Comment (dassaf4) [inline core/geometry/src/geometry3d/Matrix3d.ts:2564]: Typo in last coffs index

File: core/geometry/src/geometry3d/Matrix3d.ts [hunk 4]
```diff
@@ -2525,86 +2552,105 @@ export class Matrix3d implements BeJSONFunctions {
...
+    if (Math.sqrt(sumOff) <=
+      Geometry.smallAngleRadians * (1.0 + Math.sqrt(sumAll))
+      && Geometry.isSameCoordinate(this.coffs[0], this.coffs[4])
+      && Geometry.isSameCoordinate(this.coffs[0], this.coffs[8])
+    )
       return this.coffs[0];
     return undefined;
   }
-  /** Sum of squared differences between symmetric pairs */
+  /** Sum of squared differences between symmetric pairs (entry 1 and 3 - 2 and 6 - 5 and 7) */
```
Comment (dassaf4) [inline core/geometry/src/geometry3d/Matrix3d.ts:2583]: The dashes confused me as minus signs. Consider a new sentence: "Symmetric pairs have indices (1,3), (2,6), and (5,7)."

**Review by dassaf4 — APPROVED** [review]

File: core/geometry/src/Geometry.ts [hunk 5]
```diff
@@ -701,18 +704,40 @@ export class Geometry {
   }
   /** return 0 if the value is undefined, 1 if defined. */
   public static defined01(value: any): number { return value === undefined ? 0 : 1; }
-  /** normally, return numerator/denominator.
-   * but if the ratio would exceed Geometry.largeFractionResult, return undefined.
+  /**
+   * Return `numerator` over `denominator` or `undefined`.
```
[the remark this answers is not in the document]
Comment (saeeedtorabi) [reply core/geometry/src/Geometry.ts:708]: will fix

File: core/geometry/src/geometry3d/Matrix3d.ts (hunk 3 above)
Comment (saeeedtorabi) [reply core/geometry/src/geometry3d/Matrix3d.ts:2564]: already fixed

File: core/geometry/src/geometry3d/Matrix3d.ts (hunk 4 above)
Comment (saeeedtorabi) [reply core/geometry/src/geometry3d/Matrix3d.ts:2583]: will fix

**Pull request merged**